test/opt/graphics_robust_access_test.cpp

```diff
@@ -1548,6 +1548,184 @@ OpFunctionEnd
+               OpFunctionEnd
+)";
+
+  std::string expected = R"(OpCapability Shader
+%1 = OpExtInstImport "GLSL.std.450"
+OpMemoryModel Logical GLSL450
+OpEntryPoint GLCompute %main "else" %gl_GlobalInvocationID
+OpExecutionMode %main LocalSize 1 1 3338665985
+OpSource GLSL 450
+OpSourceExtension "GL_GOOGLE_cpp_style_line_directive"
+OpSourceExtension "GL_GOOGLE_include_directive"
+OpName %main "main"
+OpName %index "index"
+OpName %gl_GlobalInvocationID "gl_GlobalInvocationID"
+OpName %S "S"
+OpMemberName %_struct_6 0 ""
+OpMemberName %_struct_6 1 ""
+OpName %Dst "Dst"
+OpMemberName %Dst 0 "s"
+OpName %dst "dst"
+OpName %Src "Src"
+OpMemberName %Src 0 "s"
+OpName %src "src"
+OpDecorate %gl_GlobalInvocationID BuiltIn GlobalInvocationId
+OpMemberDecorate %_struct_6 0 Offset 64
+OpMemberDecorate %_struct_6 1 Offset 8
+OpDecorate %_arr__struct_6_uint_1 ArrayStride 16
+OpMemberDecorate %Dst 0 Offset 0
+OpDecorate %Dst BufferBlock
+OpDecorate %dst DescriptorSet 0
+OpDecorate %dst Binding 1
+OpDecorate %_arr__struct_6_uint_1_0 ArrayStride 16
+OpMemberDecorate %Src 0 Offset 0
+OpDecorate %Src Block
+OpDecorate %src DescriptorSet 0
+OpDecorate %src Binding 0
+%void = OpTypeVoid
+%14 = OpTypeFunction %void
+%uint = OpTypeInt 32 0
+%_ptr_Function_uint = OpTypePointer Function %uint
+%v3uint = OpTypeVector %uint 3
+%_ptr_Input_v3uint = OpTypePointer Input %v3uint
+%gl_GlobalInvocationID = OpVariable %_ptr_Input_v3uint Input
+%uint_4864 = OpConstant %uint 4864
+%_ptr_Input_uint = OpTypePointer Input %uint
+%uint_1 = OpConstant %uint 1
+%bool = OpTypeBool
+%v2uint = OpTypeVector %uint 2
+%_struct_6 = OpTypeStruct %_ptr_Input_uint %v2uint
+%_arr__struct_6_uint_1 = OpTypeArray %_struct_6 %uint_1
+%Dst = OpTypeStruct %_arr__struct_6_uint_1
+%_ptr_Uniform_Dst = OpTypePointer Uniform %Dst
+%dst = OpVariable %_ptr_Uniform_Dst Uniform
+%int = OpTypeInt 32 1
+%int_0 = OpConstant %int 0
+%_arr__struct_6_uint_1_0 = OpTypeArray %_struct_6 %uint_1
+%Src = OpTypeStruct %_arr__struct_6_uint_1_0
+%_ptr_Uniform_Src = OpTypePointer Uniform %Src
+%src = OpVariable %_ptr_Uniform_Src Uniform
+%_ptr_Uniform__struct_6 = OpTypePointer Uniform %_struct_6
+%int_2 = OpConstant %int 2
+%uint_0 = OpConstant %uint 0
+%main = OpFunction %void None %14
+%29 = OpLabel
+%index = OpVariable %_ptr_Function_uint Function
+%30 = OpAccessChain %_ptr_Input_uint %gl_GlobalInvocationID %int_2
+%31 = OpLoad %uint %30
+OpStore %index %31
+%32 = OpLoad %uint %index
+%S = OpUGreaterThanEqual %bool %32 %uint_1
+OpSelectionMerge %33 None
+OpBranchConditional %S %34 %33
+%34 = OpLabel
+OpReturn
+%33 = OpLabel
+%35 = OpLoad %uint %index
+%36 = OpLoad %uint %index
+%37 = OpAccessChain %_ptr_Uniform__struct_6 %src %int_0 %uint_0
+%38 = OpLoad %_struct_6 %37
+%39 = OpAccessChain %_ptr_Uniform__struct_6 %dst %int_0 %uint_0
+OpStore %39 %38
+OpReturn
+OpFunctionEnd
+)";
+
+  SinglePassRunAndCheck<GraphicsRobustAccessPass>(shader, expected, false,
+                                                  true);
```

Comment:
We tried to move away from this style of test because it is so hard to know what changed between "shader" and "expected", and to know which part of expected is actually important.  Could you modify this to use a match test.  Here is an example: https://github.com/KhronosGroup/SPIRV-Tools/blob/22b82872b4892b88817075be9308b97d77ee0276/test/opt/scalar_replacement_test.cpp#L28.

The idea is that the "CHECK" statements help the important changes standout, and they can be more robust because they can match a regular expression.

Reply:
I don't really know what's important here - that wasn't the point of the test. I could inline the call to `SinglePassRunAndCheck` so that we just check the returned `status` and ignore the output of the transform?

Reply:
Changed to just call `SinglePassRunToBinary<GraphicsRobustAccessPass>` and check the status code.